### DISCUSSION TITLE
load JavaScript on demand only once

### DISCUSSION THREAD

**Discussion by philippejadin** [unpoly/unpoly#516] [Ideas]
Now that unpoly execute again script tags in fragments, for a truly unobtrusive approach (one where you can add it remove unpoly and everything still works), it would be cool to be able to load js library on demand, but only once.

For example the script tags could have a up-once attribute that would tell unpoly to load the library references in SRC but only do it once for the life of the app. Can I it be done technically?

**Comment by triskweline** [comment]
Try loading your scripts with `<script type="module" src="...">`. The browser should load each URL only once, even if there are multiple script elements with that URL.

**Comment by philippejadin** [comment]
This is a very cool solution, I will try! If it's reliable, might be a good idea to add it to the docs. Thank you and thank you again for unpoly
